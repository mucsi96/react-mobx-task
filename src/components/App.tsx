import * as React from 'react';
import { inject, observer } from 'mobx-react';
import DictionaryList from './DictionaryList';
import ViewStore from '../stores/ViewStore';
import { ViewName } from '../models/View';

@inject('viewStore')
@observer
export default class App extends React.Component {
  get injected() {
    return this.props as { viewStore: ViewStore };
  }

  render(): JSX.Element {
    switch (this.injected.viewStore.currentView.name) {
      case ViewName.Overview:
        return <DictionaryList />;
      case ViewName.Dictionary:
        return <span>{'Dictionary'}</span>;
      default:
        return <span>{'Not found'}</span>;
    }
  }
}
