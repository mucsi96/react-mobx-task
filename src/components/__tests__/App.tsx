import * as React from 'react';
import * as RTL from 'react-testing-library';
import App from '../App';
import { Provider } from 'mobx-react';

const viewStore = { currentView: { render: () => 'CURRENT VIEW' } };

const render = () =>
  RTL.render(
    <Provider viewStore={viewStore}>
      <App />
    </Provider>
  );

describe('App', () => {
  it('renders', () => {
    const { asFragment } = render();
    expect(asFragment()).toMatchSnapshot();
  });
});
