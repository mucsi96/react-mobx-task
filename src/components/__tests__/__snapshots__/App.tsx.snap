// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <div
    class="ui fixed inverted menu"
  >
    <div
      class="ui container"
    >
      <a
        class="header item"
        href="/"
      >
        Transformation dictionaries manager
      </a>
    </div>
  </div>
  <div
    class="ui main text container"
  >
    CURRENT VIEW
  </div>
</DocumentFragment>
`;
